app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -546,6 +564,31 @@ class SettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    private fun setupUseLocalFxaServerPreference() {
+        val useLocalFxAServer = getPreferenceKey(R.string.pref_key_use_local_fxa_server)
+        val preferenceUseLocalFxAServer = findPreference<SwitchPreference>(useLocalFxAServer)
+        val visible = Config.channel.isMozillaOnline
+
+        preferenceUseLocalFxAServer?.apply {
+            isVisible = visible
+        }
+
+        preferenceUseLocalFxAServer?.onPreferenceChangeListener =
```

Comment:
You can short-circuit if this isn't a `isMozillaOnline` build. No need to setup `onPreferenceChangeListener` here in that case.